AElf.OS.Network.Grpc/GrpcNetworkServer.cs

```diff
@@ -0,0 +1,207 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using AElf.Common;
+using AElf.Kernel.Account;
+using AElf.OS.Network.Grpc.Events;
+using AElf.OS.Network.Temp;
+using Google.Protobuf;
+using Grpc.Core;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Org.BouncyCastle.Utilities.Encoders;
+using Volo.Abp.DependencyInjection;
+using Volo.Abp.EventBus.Local;
+
+namespace AElf.OS.Network.Grpc
+{
+    public class GrpcNetworkServer : IAElfNetworkServer, IPeerAuthentificator, ISingletonDependency
+    {
+        private readonly NetworkOptions _networkOptions;
+        
+        private readonly IAccountService _accountService;
+        private readonly IBlockService _blockService;
+        
+        private Server _server;
+        private readonly List<GrpcPeer> _authenticatedPeers;
+        
+        public ILocalEventBus EventBus { get; set; }
+        public ILogger<GrpcNetworkServer> Logger { get; set; }
+        
+        public GrpcNetworkServer(IOptionsSnapshot<NetworkOptions> options, IAccountService accountService, 
+            IBlockService blockService)
+        {
+            _accountService = accountService;
+            _blockService = blockService;
+            _networkOptions = options.Value;
+            
+            Logger = NullLogger<GrpcNetworkServer>.Instance;
+            EventBus = NullLocalEventBus.Instance;
+            
+            _authenticatedPeers = new List<GrpcPeer>();
+        }
+        
+        public async Task StartAsync()
+        {
+            // todo inject block service
+            var p = new GrpcPeerService(this, _blockService);
+            p.EventBus = EventBus;
+            
+            p.PeerSentDisconnection += POnPeerSentDisconnection;
+            
+            _server = new Server {
+                Services = { PeerService.BindService(p) },
+                Ports = { new ServerPort(IPAddress.Any.ToString(), _networkOptions.ListeningPort, ServerCredentials.Insecure) }
+            };
+            
+            _server.Start();
+            
+            // Add the provided boot nodes
+            if (_networkOptions.BootNodes != null && _networkOptions.BootNodes.Any())
+            {
+                foreach (var btn in _networkOptions.BootNodes)
+                {
+                    await Dial(btn);
```

Comment:
maybe Task.WaitAll(_networkOptions.BootNodes.Select(btn => Dial(btn)).ToArray())

Reply:
if one timeout 2s, 10 boots we need to wait 20s, but parallel connection can keep wait only 2s.